aquadoggo/src/network/behaviour.rs

```diff
@@ -166,29 +130,104 @@ impl Behaviour {
         // flag is set
         let relay_server = if network_config.relay_server_enabled {
             debug!("Relay server network behaviour enabled");
-            Some(relay::Behaviour::new(peer_id, relay::Config::default()))
+            Some(relay::Behaviour::new(
+                peer_id,
+                relay::Config {
+                    max_circuit_duration: Duration::from_secs(60 * 60),
+                    reservation_duration: Duration::from_secs(60 * 60),
+                    ..relay::Config::default()
+                },
+            ))
+        } else {
+            None
+        };
+
+        let dcutr = if network_config.relay_server_enabled || relay_client.is_some() {
+            Some(dcutr::Behaviour::new(peer_id))
         } else {
             None
         };
 
         // Create behaviour to manage peer connections and handle p2panda messaging
         let peers = peers::Behaviour::new();
 
-        // Create a behaviour to dial discovered peers.
-        let dialer = dialer::Behaviour::new();
-
         Ok(Self {
-            autonat: autonat.into(),
             identify: identify.into(),
+            mdns: mdns.into(),
             limits,
-            mdns: mdns.into(), // Convert the `Option` into a `Toggle`
-            ping: ping.into(),
             rendezvous_client: rendezvous_client.into(),
             rendezvous_server: rendezvous_server.into(),
             relay_client: relay_client.into(),
             relay_server: relay_server.into(),
+            dcutr: dcutr.into(),
             peers,
-            dialer,
         })
     }
 }
+
+#[derive(Debug)]
+pub enum Event {
```

Comment:
@sandreae, I remember you mentioned some weeks ago why this wrapper enum was required. Was it when we have two swarms? Is this still necessary?

Reply:
This is a pattern which I copied cos I saw it in many other projects using libp2p, and also their own examples. To be honest, I didn't think deeply about it 😅

Reply:
I've just played with it, don't think it's necessary, but it makes the event types look nicer!